src/mesh/mesh.ts

```diff
@@ -173,4 +174,13 @@ export class Mesh3D extends Container3D {
   static createPlane(material: Material = new StandardMaterial()) {
     return new Mesh3D(PlaneGeometry.create(), material)
   }
+
+  /**
+   * Creates a new sphere mesh with the specified material.
+   * @param material The material to use.
+   */
+  static createSphere(material?: Material, radius = 1, widthSegments = 32, heightSegments = 16, phiStart = 0, phiLength = Math.PI * 2, thetaStart = 0, thetaLength = Math.PI) {
```

Comment:
I think it would be better if everything after material would be inside an options object instead. Also add docs for those options.

Reply:
An interface could be created for that options object.

Reply:
added jsdocs
disagree on dedicated options object for this case (interfaces less intuitive for multiple params with non-zero defaults, extra init needed etc).

Reply:
What do you mean extra init needed? The options object should have the same defaults as the args would.

Reply:
Show example of non-zero default value Options object within this library.

Reply:
Something like this using object spread with default values:

```
interface DoSomethingOptions {
  a?: number,
  b?: string
}

function doSomething(options: DoSomethingOptions = {}) {
  let {a = 1, b = "something"} = options
}
```

Reply:
That's what I mean by an extra init needed.

I see this approach is used in one other place in the library with non-zero values (PostProcessingSprite constructor).  Whereas I also see constructors that adopt params with defaults (e.g. Color) so there is inconsistent implementation.

In this case (a factory method), I don't see value in replacing something solved at language level (default params in a method) with a proprietary approach (options object with init).  Factory methods are already expressive helpers.

Drawbacks to consider: more work to use, more definitions to maintain, more work to port functionality from Three etc, proprietary api style.

Can we agree this is an unnecessary burden?

Reply:
It's harder to understand this code:

`createSphere(material, 1, 32, 16, 0, Math.PI * 2, 0, Math.PI * 2)`

or this:

`createSphere(material, undefined, undefined, undefined, undefined, undefined, undefined, Math.PI * 2)`

to this:

`createSphere(material, {thetaLength: Math.PI * 2})`

This is the problem when having a large number of arguments in a function. This is hardly a proprietary solution, even PixiJS itself uses this in some places. https://pixijs.download/dev/docs/PIXI.DisplayObject.html#destroy for example.

I don't think an options object should be used for required args though.

Reply:
Harder to understand?  Maybe, maybe not.  Depends how one codes or what IDE one uses.  For example I don't use JS or TS, so it's all moot to me.  I recall many of the Pixi methods have overloading to allow params or options as wanted.

Feel free to enhance / implement it this way if you like, but it's not something I care to spend time on.

I'd rather spend said time porting larger chunks of Three (if adopting Pixi3d).  So we wouldn't want the burden of redesigning Three's interface (which has high adoption) for what is largely a decorative rationalisation on this interface (which has low adoption).  Not intended as a criticism of adoption, just utility of cross usage of tutorials, examples etc.